src/rpc/rawtransaction.cpp

```diff
@@ -1421,10 +1425,14 @@ UniValue signrawtransaction(const JSONRPCRequest& request)
         if (!txin.m_is_pegin && (coins == NULL || !coins->IsAvailable(txin.prevout.n))) {
             TxInErrorToJSON(txin, vErrors, "Input not found or already spent");
             continue;
-        } else if (txin.m_is_pegin && (txConst.wit.vtxinwit.size() <= i || !IsValidPeginWitness(txConst.wit.vtxinwit[i].m_pegin_witness, txin.prevout))) {
+        } else if (txin.m_is_pegin && (txConst.wit.vtxinwit.size() <= i || !IsValidPeginWitness(txConst.wit.vtxinwit[i].m_pegin_witness, txin.prevout, false))) {
             TxInErrorToJSON(txin, vErrors, "Peg-in input has invalid proof.");
             continue;
         }
+        // Report warning about immature peg-in though
+        if(!IsValidPeginWitness(txConst.wit.vtxinwit[i].m_pegin_witness, txin.prevout)) {
```

Comment:
I would add the `, true` here to make the difference more clear with the earlier call, but I can add that in the rebase, it's not critical.

Reply:
you can probably just remove the optional part of the argument in the rebase